tensorflow/compiler/xla/service/gpu/cudnn_pad_for_convolutions.h

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_COMPILER_XLA_SERVICE_GPU_CUDNN_PAD_FOR_CONVOLUTIONS_H_
+#define TENSORFLOW_COMPILER_XLA_SERVICE_GPU_CUDNN_PAD_FOR_CONVOLUTIONS_H_
+
+#include "tensorflow/compiler/xla/literal_util.h"
+#include "tensorflow/compiler/xla/service/gpu/ir_emission_utils.h"
+#include "tensorflow/compiler/xla/service/hlo_casting_utils.h"
+#include "tensorflow/compiler/xla/service/hlo_pass_interface.h"
+#include "tensorflow/compiler/xla/util.h"
+#include "tensorflow/compiler/xla/window_util.h"
+
+namespace xla {
+namespace gpu {
+
+// Two zero-paddings for CuDNN thunking are done in this transform: padding for
+// tensor cores and padding for integer convolutions.  This transform also
+// add slice instruction to remove unnecessary output features.
+class CudnnPadForConvolutions : public HloModulePass {
+ public:
+  explicit CudnnPadForConvolutions(bool is_volta_or_later)
```

Comment:
One review comment from the previous PR is to let CudnnPadForConvolutions take StreamExecutor*. 
It makes HLO unit test difficult.  I am not sure whether it is possible for a HLO unit test to pass a StreamExecutor to construct a CudnnPadForConvolutions object.  Please see a use case in cudnn_pad_for_convolutions_test.cc @ line 46.

Reply:
Regarding your comment in the old PR, take a look at gemm_rewrite_test.cc and `default_executor()`. Then it should be easy for `Run()` to take a `StreamExecutor*`.

Reply:
The default_executor has both major and minor versions set to -1.  I don't know how to set versions to pretend Volta.  The tests copied from cudnn_conv_pad_for_tensor_cores_test however are only for Volta.

Reply:
Hmm let's not dig into this too deeply, as it's not a big deal. Keeping the bool is good.